src/query/storages/fuse/src/operations/read_partitions.rs

```diff
@@ -104,6 +105,14 @@ impl FuseTable {
         );
         match snapshot {
             Some(snapshot) => {
+                let selected_segment = ctx.get_selected_segment_locations();
```

Comment:
I am lost here, seems selected segment locations are set into ctx in `RealHilbertClusteringHandler::do_hilbert_clustering`, why segments collected while clustering table data  are related to `Table::read_partitions` directly?

Reply:
To optimize the Hilbert clustering logic, it is necessary to pre-set the segments for reclustering. Since the recluster logic requires scanning the table twice, fetching the segments directly during read_partitions can avoid redundant selection logic and ensure that the same data is accessed during both scans.

Reply:
ok I see, let's refactor this later if necessary/possible